src/Simplify_Mul.cpp

```diff
@@ -79,6 +79,12 @@ Expr Simplify::visit(const Mul *op, ExprInfo *bounds) {
             rewrite(x * (y * c0), (x * y) * c0) ||
             rewrite(max(x, y) * min(x, y), x * y) ||
             rewrite(max(x, y) * min(y, x), y * x) ||
+
+            rewrite(min(x * c0, c1) * c2, min(x * fold(c0 * c2), fold(c1 * c2)), c2 > 0) ||
```

Comment:
I'm not comfortable with rules that makes constants larger, because it risks triggering signed integer overflow, and we're not completely sound with respect to that.

Reply:
In one of the above comments, you mentioned that a rule will fail to match if it detects overflow - so is this specifically not wanting to produce larger constants with multiplication? Regardless, I removed these in a827ae5

Reply:
It also appears that the second of these rules was causing the fuzz_simplify failure on the buildbots of 0a00af4 , so the lack of comfort with these rules seems plenty reasonable.